ambry-router/src/test/java/com.github.ambry.router/GetBlobOperationTest.java

```diff
@@ -1215,6 +1212,9 @@ public void run() {
                   chunksLeftToRead--;
                 }
                 result.getBlobResult.getBlobDataChannel().close();
+                while (writableChannel.getNextChunk(100) != null) {
```

Comment:
In which case, there might be a chunk left? I feel like chunks have been read in previous loop.

Reply:
They are not all read in the previous loop since this test case is to test when we close the ReadableChannel before we finishing read all the data from it, as you can see we close the channel before this line. But the AsyncWritableChannel's interface guarantees that whenever you pass a callback to the write method, this callback will be invoked in the same order they are passed to the channel. So this loop is doing exactly that. It loops over all the left  data chunk in the asyncWritableChannel and calls those callbacks.

Reply:
I see. I thought `chunksLeftToRead` was the total number of chunks that written into the channel.